provider/cassandra.go

```diff
@@ -0,0 +1,229 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gocql/gocql"
+	sn "github.com/mrz1836/go-sanitize"
+)
+
+type cassandraTableKey struct {
+	Keyspace, Feature, Variant string
+}
+
+func (t cassandraTableKey) String() string {
+	marshalled, err := json.Marshal(t)
+	if err != nil {
+		return err.Error()
+	}
+	return string(marshalled)
+}
+
+type cassandraOnlineStore struct {
+	session  *gocql.Session
+	keyspace string
+	BaseProvider
+}
+
+type cassandraOnlineTable struct {
+	session   *gocql.Session
+	key       cassandraTableKey
+	valueType ValueType
+}
+
+func cassandraOnlineStoreFactory(serialized SerializedConfig) (Provider, error) {
+	cassandraConfig := &CassandraConfig{}
+	if err := cassandraConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	if cassandraConfig.keyspace == "" {
+		cassandraConfig.keyspace = "Featureform_table__"
+	}
+
+	return NewCassandraOnlineStore(cassandraConfig)
+}
+
+func NewCassandraOnlineStore(options *CassandraConfig) (*cassandraOnlineStore, error) {
+	cassandraCluster := gocql.NewCluster(options.Addr)
+	cassandraCluster.Authenticator = gocql.PasswordAuthenticator{
+		Username: options.Username,
+		Password: options.Password,
+	}
+	err := cassandraCluster.Consistency.UnmarshalText([]byte(options.Consistency))
+	if err != nil {
+		return nil, err
+	}
+	newSession, err := cassandraCluster.CreateSession()
+	if err != nil {
+		return nil, err
+	}
+
+	query := fmt.Sprintf("CREATE KEYSPACE IF NOT EXISTS %s WITH replication = {'class' : 'SimpleStrategy','replication_factor' : %d }", options.keyspace, options.Replication)
+	err = newSession.Query(query).WithContext(ctx).Exec()
+	cassandraCluster.Keyspace = options.keyspace
+	if err != nil {
+		return nil, err
+	}
+
+	query = fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (tableName text PRIMARY KEY, tableType text)", fmt.Sprintf("%s.tableMetadata", options.keyspace))
+	err = newSession.Query(query).WithContext(ctx).Exec()
+	if err != nil {
+		return nil, err
+	}
+
+	return &cassandraOnlineStore{newSession, options.keyspace, BaseProvider{
+		ProviderType:   CassandraOnline,
+		ProviderConfig: options.Serialized(),
+	},
+	}, nil
+}
+
+func (store *cassandraOnlineStore) AsOnlineStore() (OnlineStore, error) {
+	return store, nil
+}
+
+func (store *cassandraOnlineStore) CreateTable(feature, variant string, valueType ValueType) (OnlineStoreTable, error) {
+	tableName := fmt.Sprintf("%s.table%sv%s", store.keyspace, sn.Custom(feature, "[^a-zA-Z0-9_]"), sn.Custom(variant, "[^a-zA-Z0-9_]"))
+	vType := cassandraTypeMap[string(valueType)]
+	key := cassandraTableKey{store.keyspace, feature, variant}
+	getTable, _ := store.GetTable(feature, variant)
+	if getTable != nil {
+		return nil, &TableAlreadyExists{feature, variant}
+	}
+
+	metadataTableName := fmt.Sprintf("%s.tableMetadata", store.keyspace)
```

Comment:
Same here, make a 
GetMetadataTableName
and call that each time so its all in one place